OrbitVulkanLayer/include/OrbitVulkanLayer/LayerLogic.h

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_LAYER_LAYER_LOGIC_H_
+#define ORBIT_LAYER_LAYER_LOGIC_H_
+
+#include <OrbitBase/Logging.h>
+
+#include "OrbitVulkanLayer/CommandBufferManager.h"
+#include "OrbitVulkanLayer/DispatchTable.h"
+#include "OrbitVulkanLayer/QueryManager.h"
+#include "QueueManager.h"
+#include "vulkan/vulkan.h"
+
+namespace orbit::layer {
+
+/**
+ * This class controls the logic of this layer. For the instrumented vulkan functions,
+ * it provides PreCall*, PostCall* and Call* functions, where the Call* function, just forward
+ * to the next layer (using the dispatch table).
+ * PreCall* functions are executed before the `actual` vulkan call and PostCall* afterwards.
+ * PreCall/PostCall are omitted when not needed.
```

Comment:
It depends on our logic. I am still undecided if we want to include them or not. For now, I would suggest to omit them to not have additional calls (not that it would matter).